base/pull_serializer_body.hpp

```diff
@@ -117,18 +132,39 @@ inline Bytes PullSerializer::Pull() {
   return result;
 }
 
-inline Bytes PullSerializer::Push(Bytes const bytes) {
+inline Bytes PullSerializer::Push(Bytes bytes) {
   Bytes result;
   CHECK_GE(chunk_size_, bytes.size);
+  if (bytes.size > 0 && compressor_ != nullptr) {
+    Bytes compressed_bytes;
+    {
+      std::unique_lock<std::mutex> l(lock_);
+      CHECK_LE(1u + number_of_compression_chunks_, free_.size());
```

Comment:
Let's not use unsigned literals to induce promotion, it confuses me. Same below.

Reply:
Done.